### PR TITLE
chore: remove unused lens implementations

### LINKED ISSUE #649: Deprecate Lens abstraction & replace with interface backed by Lily
Replacing the Lens abstraction with a lily-back interface will have the following positive impacts:
- Reduce code complexity and edge-case handling. e.g. some tasks require lily, lens opener pattern isn't needed with lily
- Reduce documentation requirements. We just need to document how to run Visor with lily.
- Ease visor usability since there is a single way to run visor via lily + performance improvements since lily is the fastest way to run visor.

The repo lens functionality can be mimicked by starting lily with bootstrapping disabled.

### PR DESCRIPTION
**Pull request by frrist**
- remove lens opener patterns
- remove vectors (as they require a removed lens)
- remove some unused flags
- closes #649

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -131,16 +131,6 @@ jobs:
       - run: sudo apt-get update
       - run: make deps
       - restore_cache:
-          name: Restore test vector files
```
Comment (frrist) [inline .circleci/config.yml:134]: Tracking via https://github.com/filecoin-project/sentinel-visor/issues/651

File: chain/filter.go [hunk 2]
```diff
@@ -1,13 +0,0 @@
-package chain
```
Comment (frrist) [inline chain/filter.go:1]: was only used via vectors.

File: chain/indexer.go [hunk 3]
```diff
@@ -113,31 +103,28 @@ func NewTipSetIndexer(o lens.APIOpener, d model.Storage, window time.Duration, n
...
+			tsi.actorProcessors[ActorStatesVerifreg] = actorstate.NewTask(node, actorstate.NewTypedActorExtractorMap(verifreg.AllCodes()))
 		case MultisigApprovalsTask:
-			tsi.messageProcessors[MultisigApprovalsTask] = msapprovals.NewTask(o)
+			tsi.messageProcessors[MultisigApprovalsTask] = msapprovals.NewTask(node)
 		case ChainConsensusTask:
 			tsi.consensusProcessor[ChainConsensusTask] = consensus.NewTask()
 		case ImplicitMessageTask:
-			if !o.Daemonized() {
-				return nil, xerrors.Errorf("daemonized API (lily node) required to run: %s task", ImplicitMessageTask)
-			}
```
Comment (frrist) [inline chain/indexer.go:140]: tasks accept API instead of opener

File: chain/indexer.go [hunk 4]
```diff
@@ -256,13 +234,6 @@ func (t *TipSetIndexer) TipSet(ctx context.Context, ts *types.TipSet) error {
 							}
 							if err == nil {
 								ll.Debugw("found actor state changes", "count", len(changes), "time", time.Since(changesStart))
-								if t.addressFilter != nil {
-									for addr := range changes {
-										if !t.addressFilter.Allow(addr) {
-											delete(changes, addr)
-										}
-									}
-								}
```
Comment (frrist) [inline chain/indexer.go:265]: removed as no longer used (was only used by vector tests)

File: chain/indexer.go [hunk 5]
```diff
@@ -389,9 +360,6 @@ func (t *TipSetIndexer) TipSet(ctx context.Context, ts *types.TipSet) error {
 		if res.Error != nil {
 			llt.Errorw("task returned with error", "error", res.Error.Error())
 			// tell all the processors to close their connections to the lens, they can reopen when needed
-			if err := t.closeProcessors(); err != nil {
```
Comment (frrist) [inline chain/indexer.go:392]: task no longer use opener pattern so they do not need to be closed. The TipSetIndexer **does** still have a close method to deal with the persistence slot, which is unchanged.

File: chain/indexer.go [hunk 6]
```diff
@@ -550,9 +518,7 @@ func (t *TipSetIndexer) stateChangedActors(ctx context.Context, old, new cid.Cid
 		return nil, err
 	}
 
-	// efficient HAMT diffing does not work over the API lens
-	_, isLotusAPILens := t.node.(*lotus.APIWrapper)
-	if !isLotusAPILens && newVersion == oldVersion && (newVersion != types.StateTreeVersion0 && newVersion != types.StateTreeVersion1) {
+	if newVersion == oldVersion && (newVersion != types.StateTreeVersion0 && newVersion != types.StateTreeVersion1) {
```
Comment (frrist) [inline chain/indexer.go:521]: we no longer support the lotus API lens so this is removed.

File: commands/client.go [hunk 7]
```diff
@@ -1,32 +0,0 @@
-package commands
```
Comment (frrist) [inline commands/client.go:1]: moved to daemon file

File: commands/visorcmd_opts.go [hunk 8]
```diff
@@ -1,31 +0,0 @@
-package commands
```
Comment (frrist) [inline commands/visorcmd_opts.go:1]: I broke this up and moved it to setup.go.

**Comment by iand** [comment]
Haven't reviewed yet but let's make sure we tag a release before this is merged

**Comment by iand** [comment]
Here is an issue that I encountered today that I have deferred fixing until after this change. Linking here to remind us. https://github.com/filecoin-project/sentinel-visor/issues/659

**Review by iand — APPROVED** [review]

**Pull request merged**